lib/api/user_repo.dart

```diff
@@ -1,19 +1,48 @@
 import 'package:flutter/foundation.dart';
+import 'package:graphql/client.dart';
 import 'package:stronk/api/graphql.dart';
 import 'package:stronk/domain/model/user.dart';
 
 abstract class UserRepository {
   Future<User> retrieveUser();
+
+  Future<User> retrieveUserById(String id);
 }
 
 class UserRepositoryImpl implements UserRepository {
   GraphQLUtility utility;
 
-  UserRepositoryImpl({@required GraphQLUtility utility});
+  UserRepositoryImpl({@required this.utility});
 
   @override
   Future<User> retrieveUser() async {
     // todo replace with real user fetch call
-    return User(name: "Test user");
+    return User(name: "test");
   }
-}
\ No newline at end of file
+
+  @override
+  Future<User> retrieveUserById(String id) async {
+    const String readUsers = """
```

Comment:
let's create a class in api/graphql called `userDocument` or something and try move this into i. you can see an example here `lib/api/graphql/auth.dart` :)